config/initializers/default_host.rb

```diff
@@ -1,13 +1,15 @@
 Rails.application.config.to_prepare do
-  begin
-    if Rails.application.routes.default_url_options[:host].blank? and AppSetting.default_host_exists?
-      host = AppSetting.default_host
-      Rails.logger.info "[SUCCESS] Setting default_url_options[:host] = '#{host}'"
-      Rails.application.routes.default_url_options[:host] = host
+  unless ENV["SECRET_KEY_BASE"] == "DUMMY"
```

Comment:
I'd love to eventually address this properly but — because we make an initializer try to touch the database 😱 it has to be accounted for.

Reply:
Is this condition just a proxy for "has been setup already?" I'm having a hard time figuring out when and why this would be run.

Reply:
It's part of the song and dance happening in the Dockerfile:

https://github.com/themarshallproject/klaxon/blob/e8d394404b3fa31d1c00a0ab4e1e8e49dbcdf896/Dockerfile#L54-L55

This prevents the attempt to attach to the database during the image build (when it would never exist) just to call `assets:precompile` so the image can bundle those assets. I think normally all the `DUMMY` value is meant to do is allow `assets:precompile` to run in production mode when a real secret key is not needed yet.

Reply:
What's funny is in the Rails 7.1 "native" version of this approach they actually created a new environment variable just for this:

https://github.com/rails/rails/blob/6adaeb864946718f71746958d3463e5c0a266b30/railties/lib/rails/generators/rails/app/templates/Dockerfile.tt#L62-L63

I guess recognizing `SECRET_KEY_BASE_DUMMY` is set is easier than unpacking how `assets:precompile` has no need for the secret key.